lib/BatchTopicDistribution.js

```diff
@@ -123,4 +123,19 @@ BatchTopicDistribution.prototype.download = function (resource, filename, cb) {
   return this.connection.download(reqOptions, filename, undefined, cb);
 };
 
+BatchTopicDistribution.prototype.createAndWait = function (
+  topicModel, dataset, args, retry, cb) {
+  var self = this;
+  if (arguments.length < 5 &&
+      (typeof arguments[arguments.length - 1] === "function")) {
+    retry = false;
+    cb = arguments[arguments.length - 1];
+  }
+  self.create(topicModel, dataset, args, retry,
+    function(error, data) {
+      return self.get(data.resource, true, undefined, cb);
+    });
+    return;
+}
+
```

Comment:
have you considered a wait function somewhere in a parent class or
so instead of repeating very similar code so often?

Reply:
I did, but this method calls the create method, which is defined in the subclass as well as in the base class, so I think that, in order to use the correct method, we need to define it in the subclass too. Also, the number of parameters is different for some of the subclasses, like evaluations, predictions, etc.